command.c

```diff
@@ -5,33 +5,31 @@
 
 #include "nsq.h"
 
-static const int MAX_BUF_SIZE = 128;
-static const int MIN_BUF_SIZE = 64;
-static const int BUF_DELTER   = 2;
+static const int defaultBufSize = 32;
+static const int restBufSize    = 16;
+static const int bufDelter      = 1;
 
-void *nsq_buf_malloc(size_t buf_size, size_t n, size_t l)
+void *nsq_buf_malloc(size_t *buf_size, size_t n, size_t l)
 {
-    if (buf_size - n >= MIN_BUF_SIZE || buf_size - l >= MIN_BUF_SIZE) {
+    int realLeftSize = (int)(*buf_size - n - l);
```

Comment:
because `*buf_size`, `n`, and `l` are all `size_t` which is unsigned, it will wrap-around to a large positive value if `l > buf_size - n`

Reply:
(you have to cast to `int` before the subtraction, but here you cast only the final result)

Reply:
It's not gonna happened. The GCC has take care of this problem.

Reply:
Hmm, you're right, the result after wrap-around is the bit-wise equivalent, so it works.